### PR TITLE
Global refactor

### PR DESCRIPTION
**Pull request by beInDev**
Big refactoring to avoid further legacy

- Removed browser compatibility (tbh nobody uses it)
- Use classes
- Modern syntax
- Unit testing
- Clearer / Cleaner code
- Maintainability

and so much more...

It's ISO with master 🎉

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,4 +4,4 @@
 # Portions Copyright (C) Philipp Kewisch, 2014-2015
 
 minify:
-	(cd lib && uglifyjs -m -r 'exports,require' sepa.js > sepa.min.js)
+	(cd lib && ../node_modules/.bin/uglifyjs -m -r 'exports,require' sepa.js > sepa.min.js)
```
Comment (DrPlop) [inline Makefile]: If we still want to use this file, this should be:

```suggestion
	(cd lib && ../node_modules/.bin/uglifyjs -m -r 'exports,require' index.js > index.min.js)
```

File: lib/utils/utils.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+/**
+ * Replace letters with numbers using the SEPA scheme A=10, B=11, ...
+ * Non-alphanumerical characters are dropped.
+ *
+ * @param value     The alphanumerical input string
+ * @return        The input string with letters replaced
+ */
+function replaceChars(value) {
+  if (typeof value !== 'string') {
+    /* eslint:disable-next-line */
```
Comment (DrPlop) [inline lib/utils/utils.js]: Is this comment still necessary ? 🤔

**Comment by ArthurMialon** [comment]
🙌

**Comment by beInDev** [comment]
@DrPlop What do you think about merging this? 😉

**Comment by DrPlop** [comment]
> @DrPlop What do you think about merging this? 😉

Yeah !

**Pull request merged**